eng/common/scripts/Helpers/Resource-Helpers.ps1

```diff
@@ -134,3 +151,35 @@ filter Remove-PurgeableResources {
 function Log($Message) {
   Write-Host ('{0} - {1}' -f [DateTime]::Now.ToLongTimeString(), $Message)
 }
+
+function Wait-PurgeableResource {
+  param (
+    [Parameter(Mandatory=$true)]
+    [scriptblock] $ScriptBlock,
+
+    [Parameter(Mandatory=$true)]
```

Comment:
Do you still need to pass thru the resource?

Reply:
Nevermind I see what you used that for now.

Reply:
It's used for printing a warning that the operation timed out, as well as optionally outputting all timed-out resources at the end so we can investigate if necessary.